src/main/resources/application.conf

```diff
@@ -298,14 +298,20 @@ mantis {
     # When we receive a block that we don't have a parent for, it means we found a fork. To resolve it, we need to query the
     # same peer for previous blocks, until we find one that we have a parent for. This setting determines the batch size
     # of such query (number of blocks returned in a single query).
-    block-resolving-depth = 20
+    block-resolving-per-request = 20
 
     # size of the list that keeps track of peers that are failing to provide us with mpt node
     # we switch them to download only blockchain elements
     fastsync-block-chain-only-peers-pool = 100
 
     # time between 2 consecutive requests to peer when doing fast sync, this is to prevent flagging us as spammer
     fastsync-throttle = 2 seconds
+
+    # limit of length of branch in chain that will be resolved while doing regular sync
+    # if branch is longer resolution will be aborted because of risk of out of memory error
+    # too long branch may also indicate malicious peer, 24 because confirmation depth in ethereum is ~12 blocks
+    # so we allow branches of 2x confirmation depth
+    branch-max-depth-resolving = 24
```

Comment:
I would also rename them, cause they may seem unrelated now. How about: `branch-resolution-batch-size` and `branch-resolution-max-depth`. Also wouldn't hurt to make them adjacent in the config file.

As for the values I would say max-depth has to be a multiple of batch-size. Otherwise we will never reach the max depth (when trying to resolve).

Finally, it would be better if the check for headers queue size (>=) would come after the attempt to resolve. Otherwise, on failed resolution, we will always be making 1 redunant requests (the downloaded headers will always be discarded because max-depth was reached)